### PR TITLE
AI transformers now get function identifiers

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/analyses/ai.h [hunk 1]
```diff
@@ -398,7 +395,8 @@ template<typename domainT>
 class concurrency_aware_ait:public ait<domainT>
 {
 public:
-  typedef typename ait<domainT>::statet statet;
+  using statet = typename ait<domainT>::statet;
```
Comment (owen-mc-diffblue) [inline src/analyses/ai.h:407]: @kroening The coding standard currently says "Prefer use of typedef instead of using" but I see you prefer `using` here. Should the coding standard be updated?

**Review by owen-mc-diffblue — APPROVED** [review]

File: src/analyses/constant_propagator.h [hunk 2]
```diff
@@ -195,17 +197,6 @@ class constant_propagator_ait:public ait<constant_propagator_domaint>
...
-  constant_propagator_ait(
-    goto_functionst::goto_functiont &goto_function,
-    const namespacet &ns,
-    should_track_valuet should_track_value = track_all_values):
-    dirty(goto_function),
-    should_track_value(should_track_value)
-  {
-    operator()(goto_function, ns);
-    replace(goto_function, ns);
-  }
```
Comment (martin-cs) [inline src/analyses/constant_propagator.h]: Why is this removed?  This seems unrelated to the change described in the commit message.

**Review by martin-cs — COMMENTED** [review]
The first commit making consistent the use of locations : yes, definitely accept.  i thought I'd already fixed this but it looks like that hasn't been merged yet.

The second commit : why?  It doesn't say why the API needs to change.

**Comment by owen-mc-diffblue** [comment]
@martin-cs I assume it is because of https://github.com/diffblue/cbmc/pull/3126

**Comment by martin-cs** [comment]
@owen-jones-diffblue : thank you; that does make more sense.

File: src/analyses/constant_propagator.h (hunk 2 above)
Comment (kroening) [reply src/analyses/constant_propagator.h]: I've put it back in, with a function identifier.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 261dfc9421b98cf7204b9e8e131a3ad46db878e3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87594310
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 04c527728948cf825488974da5e36d5c8b9b9a10).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87596523

**Review by tautschnig — APPROVED** [review]
Looks good except as noted by @owen-jones-diffblue: `using` isn't currently the recommended approach. Also, it would be nice if the commit message could please say that this is part of the larger effort that #3126 is.

**Comment by kroening** [comment]
Using vs typedef: see PR #3270

**Review by chrisr-diffblue — APPROVED** [review]
Looks reasonable to me (even if my team will need to do a bit of rebasing fixup for some of our branches). I'd echo the comments about adding a bit of extra rationale to the commit messages.

**Comment by kroening** [comment]
@chrisr-diffblue Done!

**Review by smowton — APPROVED** [review]

**Review by martin-cs — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 09b5ac19da9dcd653597d1b15a54b2c6af6c6485).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90316108
